lib/foreman_maintain/scenario.rb

```diff
@@ -158,9 +158,13 @@ def to_hash
     end
 
     def self.new_from_hash(hash)
-      scenarios = find_scenarios(:label => hash[:label])
+      condition = { :label => hash[:label] }
+      scenarios = find_scenarios(condition)
```

Comment:
What is the reason for not using non-confine variant first?

Reply:
Yeah, totally agree!! :+1: 
There is no use having confine evaluation here as we are again checking for non-confine variant. 
I was thinking too much unnecessary about this I guess :-)
I will update PR according to review comments.